### PR TITLE
En_Dns

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:306]:
```suggestion
    static s32 D_8092DE00[] = {
        8,
        8,
        9
    };
```
and then the formatter should put it on one line.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:419]:
```suggestion
void EnDns_Noop(EnDns* this, GlobalContext* globalCtx) {
}
```
Might as well get this one out of the way.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:480]:
```suggestion
    static s32 D_8092DE0C[] = {
        0,
        0,
        4,
        6
    };
```
As above.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:522]: This is an interesting one. Presumably this isn't required for matching?

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:594]:
```suggestion
s32 EnDns_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
    EnDns* this = THIS;

    this->unk_1E4[limbIndex] = *dList;
    *dList = NULL;
    return false;
}
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:596]:
```suggestion
void EnDns_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:615]:
```suggestion
        func_8092D954(this->unk_2CC, this->unk_2CE + this->actor.shape.rot.y, &this->unk_218, &this->unk_224,
```
The cast seems unnecessary since the function already takes s16.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:639]:
```suggestion
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:638]: This one's a bit more open to question, depends what the symbols end up as. But I'd be inclined to remove the comma here too, at least for now.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Dns/z_en_dns.c:522]: OR'ing the flags together? Engineer recommended me to do it in a previous PR.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:623]: Those looks like textures, not Dlists
```suggestion
    static TexturePtr D_8092DE1C[] = { D_060028E8, D_06002968, D_060029E8, D_06002968 };
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:137]:
```suggestion
    Player* player = PLAYER;
    s16 yaw = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
    Vec3f sp34;
    Vec3f sp28;

    Math_ApproachS(&this->unk_2CE, temp, 4, 0x2AA8);
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:403]: Sorry, didn't read this properly. This should be `EnDns_DoNothing` since it's an actionFunc.

**Review by AngheloAlf — APPROVED** [review]

File: include/functions.h
Comment (engineer124) [inline include/functions.h:2710]: This is fine as is if you want as all these args will be temporary, but I've been trying to change the args the other way. Anytime I see `params` in a func I'm using, I've been switching them to `arg` instead of params to make it more consistent with OoT.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:313]:
```suggestion
        this->unk_2D2++;
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:330]:
```suggestion
                this->unk_2E4 -= -(40.0f / 13.0f);
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:339]:
```suggestion
                this->unk_2E4 += -(20.0f / 13.0f);
```

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (engineer124) [inline src/overlays/actors/ovl_En_Dns/z_en_dns.c:545]:
```suggestion
    if (!func_8092CAD0(this, globalCtx) && func_8092CB98(this, globalCtx)) {
        func_8092D5E8(this, globalCtx);
        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
        func_8092C5C0(this);
    } else {
        this->actionFunc(this, globalCtx);
        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
        func_8092C934(this);
        func_8092C86C(this, globalCtx);
        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 12.0f, 0.0f, 4);
        func_8013C964(&this->actor, globalCtx, 80.0f, 40.0f, 0, this->unk_2C6 & 7);
        Actor_SetHeight(&this->actor, 34.0f);
        func_8092C6FC(this, globalCtx);
        func_8092C5C0(this);
    }
```
Assuming it matches, an else should be able to get rid of the return

**Review by engineer124 — COMMENTED** [review]
Great work! Just a few small comments

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (rylieb) [reply src/overlays/actors/ovl_En_Dns/z_en_dns.c:330]: Phew! I'm so glad I'm not the only one who does this (educated guess the fraction corresponding to a float constant). I second all of engineer's suggestions here.

Comment (engineer124) [reply src/overlays/actors/ovl_En_Dns/z_en_dns.c:330]: You can also just take floats and throw them into Wolfram Alpha. It'll give you the right fraction nearly every time without any effort :)

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Dns/z_en_dns.c:330]: The annoying thing is checking if it gives the right hex representation. We could do with a hex calculator that can do that.

File: include/functions.h
Comment (Kenix3) [reply include/functions.h:2710]: I prefer arg as well, but I'm personally not picky on undocumented code. Also there's no need for consistency to oot in undocumented code.

File: src/overlays/actors/ovl_En_Dns/z_en_dns.c
Comment (Kenix3) [reply src/overlays/actors/ovl_En_Dns/z_en_dns.c:545]: I'm personally fine with an early return.

**Review by Kenix3 — APPROVED** [review]
I'd prefer some simpler forms of the floats, but I don't think that should hold up a merge in undocumented code.

Pending +1 contributor approval.

File: include/functions.h
Comment (EllipticEllipsis) [inline include/functions.h:2090]: These are 0-indexed, so weren't the labels right before, if not the types?

**Review by engineer124 — APPROVED** [review]

**Review by rylieb — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Comment by AngheloAlf** [comment]
This PR has conflicts :c

**Pull request merged**
